android/src/main/java/ly/count/android/sdk/react/CountlyReactNative.java

```diff
@@ -749,6 +749,28 @@ public void event(ReadableArray args) {
         }
     }
 
+    @ReactMethod
+    public void recordEvent(ReadableMap args) {
+        String eventName = args.getString("n");
+
+        int eventCount = 0;
+        if (args.hasKey("c")) {
+            eventCount = args.getInt("c");
+        }
+
+        double eventSum = 0;
+        if (args.hasKey("s")) {
+            eventSum = args.getDouble("s");
+        }
+
+        HashMap<String, Object> segmentation = new HashMap<>();
+        ReadableArray segments = args.getArray("g");
+        for (int i = 0, il = segments != null ? segments.size() : 0; i < il; i += 2) {
```

Comment:
this notation reduces readibility but I guess it is fine. maybe property can be defined above of the loop?

Reply:
done